crates/matrix-sdk/src/room/common.rs

```diff
@@ -194,6 +198,247 @@ impl Common {
         Ok(response)
     }
 
+    /// Get a stream for the timeline of this `Room`
+    ///
+    /// The first stream is forward in time and second stream is backward in
+    /// time. If the `Store` used implements message caching and the
+    /// timeline is cached no request to the server is made.
+    ///
+    /// The streams make sure that no duplicated events are returned. If the
+    /// event graph changed on the server existing streams will keep the
+    /// previous event order. Streams created later will use the new event
+    /// graph, and therefore will create a new local cache.
+    ///
+    /// The forward stream will only return `None` when a gapped sync was
+    /// performed. In this case also the backward stream should be dropped,
+    /// to make the best use of the message cache.
+    ///
+    /// The backward stream returns `None` once the first event of the room was
+    /// reached. The backward stream may also return an Error when a request
+    /// to the server failed. If the error is persistent new streams need to
+    /// be created.
+    ///
+    /// With the encryption feature, messages are decrypted if possible. If
+    /// decryption fails for an individual message, that message is returned
+    /// undecrypted.
+    ///
+    /// # Examples
+    /// ```no_run
```

Comment:
why are these `no_run`?

Reply:
same as all other examples. The only difference this one won't fail because we use early returns.